pkg/addons/addons.go

```diff
@@ -137,6 +137,9 @@ func EnableOrDisableAddon(cc *config.ClusterConfig, name string, val string) err
 
 	// check addon status before enabling/disabling it
 	if isAddonAlreadySet(cc, addon, enable) {
+		if addon.Name() == "gcp-auth" {
```

Comment:
if gcp-auth addon is enabled after user is logged in to gcloud, there would be no credentials sent to the pods.

currently we are asking users to re-enable the addon to pick up the credentials. would  this break that scenario ?

Reply:
No. As noted in the PR description, we still go through the process of finding creds and mounting them in. This part only skips applying the yaml for the pods and namespace and stuff

Reply:
I've now enabled the option to have pods refreshed with new credentials if --force gets passed into `addons enable`. Updating existing pods isn't an option, since kubernetes restrict what can be directly updated on an existing pod. The other option (apart from deleting and recreating pods) would be to force the use of deployments, which would still require deleting existing pods.